tensorpack/callbacks/param.py

```diff
@@ -373,10 +371,11 @@ def _get_value_to_set(self):
             return None
 
         self.history.append(last)
+        self.observations_since_last_changed += 1
 
         if len(self.history) < self.history.maxlen or \
-                self.epoch_num - self.last_changed_epoch < self.history.maxlen:
-            # not full yet, or value have changed just now
+                self.observations_since_last_changed < self.history.maxlen:
+            # not full yet, or the k observations which lead to the last change haven't been all popped out
```

Comment:
Is it better if we just clear the history?

Reply:
Yes, it seems more concise.